pkg/store/bucket.go

```diff
@@ -1945,8 +1958,13 @@ func (s *BucketStore) LabelNames(ctx context.Context, req *storepb.LabelNamesReq
 		return nil, status.Error(codes.Unknown, errors.Wrap(err, "marshal label names response hints").Error())
 	}
 
+	names := strutil.MergeSlices(sets...)
```

Comment:
Do we think it is worth it to pass limit into `strutil.MergeSlices` and truncate there? 

```
func MergeSlices(a ...[]string) []string {
	if len(a) == 0 {
		return nil
	}
	if len(a) == 1 {
		return a[0]
	}
	l := len(a) / 2
	return mergeTwoStringSlices(MergeSlices(a[:l]...), MergeSlices(a[l:]...))
}
```

For example, if the limit is < len(a) / 2, we can skip merging the second half of slice.

Reply:
But this is not a blocker, we can do it in further prs

Reply:
Thanks Ben. I can look into it. Since this PR has gotten too big, I prefer to do in a follow up PR. :)